app/full_text_search.py

```diff
@@ -113,6 +135,26 @@ def search(repo, query_txt, max_results):
     return results
 
 
+def create_fields_for_rank(field_name, value):
+    """
+    Creates fields for ranking.
+    Args:
+        field_name: field name
+        value: field value
+    Returns:
+        array of appengine_search.TextField(name=field_name, value=value)
+           (length: REPEAT_COUNT_FOR_RANK)
+    """
+    if not value:
+        return []
+
+    fields = []
+    for x in range(REPEAT_COUNT_FOR_RANK):
+        fields.append(
```

Comment:
It's up to you, but how about putting it into a single field (instead of 5 fields) but joining with a whitespace (e.g., "yamada yamada yamada yamada yamada")? It looks a bit simpler and can avoid adding too many fields.